client/client.go

```diff
@@ -1017,16 +1045,30 @@ func (c *Client) setupDrivers() error {
 		if err != nil {
 			return err
 		}
-		c.configLock.Lock()
-		applies, err := d.Fingerprint(c.config, c.config.Node)
-		c.configLock.Unlock()
+
+		request := &cstructs.FingerprintRequest{Config: c.config, Node: c.config.Node}
+		response := &cstructs.FingerprintResponse{
```

Comment:
Would prefer if it looked like:

```
var resp cstructs.FingerprintResponse
if err := d.Fingerprint(req, &resp); err != nil {
  // Handle
}
```

You can then have methods on the response to be like:

```
func (f *FingerprintResponse) AddAttribute(attr, value string) {...}
```

Where it initializes the value.

Reply:
Good idea

Reply:
Alternatively, we could have a constructor which would initialize all of these values. I will add getter/setter methods that does initialization for each field, let me know if you think it is cleaner to just do initialization all at once in a constructor.

Reply:
Yeah the constructor could work but it is just idiomatic go to call an RPC like I showed, which is the direction we will be moving.